dfusion_rust_core/src/models/standing_order.rs

```diff
@@ -123,31 +128,67 @@ impl From<&Arc<Log>> for StandingOrder {
 }
 
 
-impl From<Entity> for StandingOrder {
-    fn from(entity: Entity) -> Self {
-        let account_id = u16::from_entity(&entity, "accountId");
-        let batch_index = U256::from_entity(&entity, "batchIndex");
-        let valid_from_auction_id = U256::from_entity(&entity, "validFromAuctionId");
-
-        StandingOrder {
-            account_id,
-            batch_index,
-            valid_from_auction_id,            
-            orders: vec![] // Orders cannot be reconstructed just from the entity
+impl From<(Entity, Vec<Entity>)> for StandingOrder {
+    fn from(entities: (Entity, Vec<Entity>)) -> Self {
+        let batch_entity = entities.0;
+        let mut order_entities_iter = entities.1.into_iter();
+        let order_ids_value = batch_entity.get("orders").expect("it should contain order ids");
+
+        if let Value::List(order_ids) = order_ids_value {
```

Comment:
nit: to avoid the if/else block and keep the indentation level by linear you can do:

`let order_ids = order_ids_value.as_list.expect("orders should be a List")`

Reply:
the cyclomatic complexity of this methods says:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2FnEMmyUp4hl1QOzovKh%2Fgiphy.gif&token=thanks" data-gitmeme-token="thanks"><img src="https://media1.giphy.com/media/nEMmyUp4hl1QOzovKh/giphy.gif" title="Created by gitme.me with /thanks"/></a>